omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -257,6 +259,118 @@ else if(identifierSourceType.equals(Identifier_Pool)){
+                }
+            }
+            updatedIdentifierSource =  save(identifierSource);
+        }
+        if(Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(uuid) instanceof IdentifierPool){
+            IdentifierPool identifierSource = (IdentifierPool) Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(uuid);
+            Object name = updateBody.get("name");
+            Object description = updateBody.get("description");
+            Object batchSize = updateBody.get("batchSize");
+            Object minPoolSize = updateBody.get("minPoolSize");
+            Object sourceUuid = updateBody.get("sourceUuid");
+            Object sequential = updateBody.get("sequential");
+            Object refillWithScheduledTask = updateBody.get("refillWithScheduledTask");
+            
+            if (sourceUuid != null && StringUtils.isNotBlank(sourceUuid.toString())) {
+                IdentifierSource poolIdentifierSource = Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(sourceUuid.toString());
+                if(poolIdentifierSource != null){
+                    identifierSource.setSource(poolIdentifierSource);
+                }
+            }
+            if (name != null && StringUtils.isNotBlank(name.toString())) {
+                identifierSource.setName(name.toString());
+            }
+            if (description != null && StringUtils.isNotBlank(description.toString())) {
+                identifierSource.setDescription(description.toString());
+            }
+            if (batchSize != null && StringUtils.isNotBlank(batchSize.toString())) {
+                if(StringUtils.isNumeric(batchSize.toString())) {
+                    identifierSource.setBatchSize(Integer.parseInt(batchSize.toString()));
+                }
+            }
+            if (minPoolSize != null && StringUtils.isNotBlank(minPoolSize.toString())) {
+                if(StringUtils.isNumeric(minPoolSize.toString())) {
+                    identifierSource.setMinPoolSize(Integer.parseInt(minPoolSize.toString()));
+                }
+            }
+            if (sequential != null && StringUtils.isNotBlank(sequential.toString())) {
+                identifierSource.setSequential(parseBoolean(sequential));
+            }
+            if (refillWithScheduledTask != null && StringUtils.isNotBlank(refillWithScheduledTask.toString())) {
+                identifierSource.setRefillWithScheduledTask(parseBoolean(refillWithScheduledTask));
+            }
+            updatedIdentifierSource =  save(identifierSource);
+        }
+        if(Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(uuid) instanceof RemoteIdentifierSource){
+            RemoteIdentifierSource identifierSource = (RemoteIdentifierSource) Context.getService(IdentifierSourceService.class).getIdentifierSourceByUuid(uuid);
```

Comment:
A couple of lines in this file are bit too long like this one

Reply:
fixed